library/x509write_csr.c

```diff
@@ -219,16 +218,13 @@ static int x509write_csr_der_internal( mbedtls_x509write_csr *ctx,
      * Note: hash errors can happen only after an internal error
      */
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
-    if( psa_hash_setup( &hash_operation, hash_alg ) != PSA_SUCCESS )
-        return( MBEDTLS_ERR_X509_FATAL_ERROR );
-
-    if( psa_hash_update( &hash_operation, c, len ) != PSA_SUCCESS )
-        return( MBEDTLS_ERR_X509_FATAL_ERROR );
-
-    if( psa_hash_finish( &hash_operation, hash, sizeof( hash ), &hash_len )
-        != PSA_SUCCESS )
-    {
-        return( MBEDTLS_ERR_X509_FATAL_ERROR );
+    if( psa_hash_compute( hash_alg,
+                          c,
+                          len,
+                          hash,
+                          PSA_HASH_MAX_SIZE,
+                          &hash_len ) != PSA_SUCCESS ) {
```

Comment:
Style: braces go on their own line.
```suggestion
                          &hash_len ) != PSA_SUCCESS )
        {
```